### ISSUE TITLE
Update Username

### ISSUE THREAD

**Issue by pixie79** [keybase/client#11716]
It would be great if you could update your username, this would allow you to keep all you associations but change name at appropriate times.

For example when you change name legally, you are still the same person and your history is the same but your name should be able to change to match.

Thanks

Mark

**Comment by lilyball** [comment]
See https://github.com/keybase/keybase-issues/issues/193

**Comment by solvaholic** [comment]
Thank you for your update in https://github.com/keybase/keybase-issues/issues/193 @lilyball 

> I'm changing my complete identity, both online and in person.

Initially I thought "Keybase proves connections between identities" and I assumed a person changing their name would want to erase those connections for anonymity. But I see you'd like to rename them for identity. I think that's quite a different problem to solve.

> ... but it's still a username I'm trying to distance myself from.

Since it seems like you'd need to create a new Keybase account rather than rename one, I wonder if, through its existing features, Keybase could support this by proving your old username is connected to your new username.

Then other users could more confidently replace your cryptographic identity.

**Comment by lilyball** [comment]
> Initially I thought "Keybase proves connections between identities" and I assumed a person changing their name would want to erase those connections for anonymity. But I see you'd like to rename them for identity.

Yes. Keybase assumes that my name is my identity. But it's not, it's just a handle on my identity. Just as I can change my Twitter name or GitHub name and link the new ones back to my identity, I should be able to change my Keybase username. I changed my name, but I didn't sever ties with people who knew me.

> Since it seems like you'd need to create a new Keybase account rather than rename one, I wonder if, through its existing features, Keybase could support this by proving your old username is connected to your new username.

Given the way Keybase works today, I think they could do exactly this but in an automated fashion. As long as the username is inextricably tied to the account, they could let me create a new account, publish a proof that the new account is a continuation of the old one under a different name, and then other people using the Keybase software would see the proof and trust the new account exactly as it did the old, including following the new account if they were following the old, republishing proofs, etc. For all intents and purposes, pretending as though it's the same account but with a username update.

Of course, this is all a bit awkward, and it really would be better if Keybase just moved away from using the username as the unique identifier of the account, instead using some other generated identifier and having the username simply as a property of the account, which can be changed like other properties. There are downsides to this approach though, such as whether they need to keep abandoned usernames reserved forever.
